cmd/jaeger/internal/all-in-one.yaml

```diff
@@ -5,7 +5,19 @@ service:
       receivers: [otlp, jaeger, zipkin]
       processors: [batch]
       exporters: [jaeger_storage_exporter]
-
+  telemetry:
+    resource:
+      service.name: jaeger
+    metrics:
+      level: detailed
+    traces:
```

Comment:
There is another problem here. The OTLP receiver endpoint is now generating traces

<img width="1753" alt="image" src="https://github.com/user-attachments/assets/c98a8761-53b9-4b9a-a264-4b7ed0b4ce76">

This is something we never allowed in Jaeger v1, because it creates an infinite recursion - collector receives a trace, generates another trace as part of receiver processing, that other trace is recursively sent to the receiver, which generates a new trace, an so on.

In Jaeger-v1 the trace processing endpoints were never instrumented for tracing, for this reason. It seems OTEL collector by default does instrument grpc/http receivers, which is a no-go, we need to find a way to stop that.

One option would be to use a custom sampler that only allows tracing on the query endpoints.

Reply:
Yeah understood the infinite loop, didn't take that into consideration that's quite a huge problem

Reply:
Does OTEL provide a custom sampler?

Reply:
Sampler is an interface, we can always create something that we need (and yes some languages already support rule-based samplers). But the challenge is that we're not initializing this telemetry providers ourselves, so how we can inject it is the question.

Reply:
Ok will brainstorm about this and see if I can come up with something and will report back to you if I have any doubts or need help understanding something.
But, before that I need to fix why the UI is a no-show for me as I am blindly making changes without the UI without proper confirmation. 😅